include/libopencm3/stm32/g0/adc.h

```diff
@@ -64,48 +64,69 @@
 
 /** ADC_CALFACT Calibration factor register */
 #define ADC_CALFACT(adc)	MMIO32((adc) + 0xB4)
-
-/** ADC_OR Option register */
-#define ADC_OR(adc)			MMIO32((adc) + 0xD0)
 /**@}*/
 
+/* --- Register definitions -------------------------------------------------------*/
+
+#define ADC_SMPR(adc)		ADC_SMPR1(adc)
+#define ADC_TR(adc)		ADC_TR1(adc)
+
+#define ADC1_ISR		ADC_ISR(ADC1)
+#define ADC1_IER		ADC_IER(ADC1)
+#define ADC1_CR			ADC_CR(ADC1)
+#define ADC1_CFGR1		ADC_CFGR1(ADC1)
+#define ADC1_CFGR2		ADC_CFGR2(ADC1)
+#define ADC1_SMPR		ADC_SMPR(ADC1)
+#define ADC1_TR			ADC_TR(ADC1)
+#define ADC1_CHSELR		ADC_CHSELR(ADC1)
+#define ADC1_DR			ADC_DR(ADC1)
+#define ADC1_CCR		ADC_CCR(ADC1)
+
+#define ADC1_AWD1TR		ADC_AWD1TR(ADC1)
+#define ADC1_AWD2TR		ADC_AWD2TR(ADC1)
+#define ADC1_AWD3TR		ADC_AWD3TR(ADC1)
+#define ADC1_AWD2CR		ADC_AWD2CR(ADC1)
+#define ADC1_AWD3CR		ADC_AWD3CR(ADC1)
+#define ADC1_CALFACT		ADC_CALFACT(ADC1)
+
 /* --- Register values -------------------------------------------------------*/
 
 /** @addtogroup adc_isr
 @{*/
 
 /** CCRDY: Channel Configuration Ready flag */
-#define ADC_ISR_CCRDY				(1 << 13)
+#define ADC_ISR_CCRDY		(1 << 13)
 
 /**@}*/
 
 /** @addtogroup adc_ier
 @{*/
 
 /** CCRDYIE: Channel Configuration Ready Interrupt enable bit */
-#define ADC_IER_CCRDYIE				(1 << 13)
+#define ADC_IER_CCRDYIE		(1 << 13)
 
 /**@}*/
 
 /** @addtogroup adc_ccr
 @{*/
 
-#define ADC_CCR_PRESC_MASK		(0xf)
-#define ADC_CCR_PRESC_SHIFT		(18)
+#define ADC_CCR_PRESC_SHIFT	18
+#define ADC_CCR_PRESC_MASK	(0xf << ADC_CCR_PRESC_SHIFT)
+#define ADC_CCR_PRESC_VAL(x)	((x) << ADC_CCR_PRESC_SHIFT)
 /** @defgroup adc_ccr_presc ADC clock prescaler
  *@{*/
-#define ADC_CCR_PRESC_NODIV		(0x0)
-#define ADC_CCR_PRESC_DIV2		(0x1)
-#define ADC_CCR_PRESC_DIV4		(0x2)
-#define ADC_CCR_PRESC_DIV6		(0x3)
-#define ADC_CCR_PRESC_DIV8		(0x4)
-#define ADC_CCR_PRESC_DIV10		(0x5)
-#define ADC_CCR_PRESC_DIV12		(0x6)
-#define ADC_CCR_PRESC_DIV16		(0x7)
-#define ADC_CCR_PRESC_DIV32		(0x8)
-#define ADC_CCR_PRESC_DIV64		(0x9)
-#define ADC_CCR_PRESC_DIV128	(0xa)
-#define ADC_CCR_PRESC_DIV256	(0xb)
+#define ADC_CCR_PRESC_NODIV	0x0
+#define ADC_CCR_PRESC_DIV1	0x1
+#define ADC_CCR_PRESC_DIV2	0x2
```

Comment:
you rebased your branch a little quickly here, you've re-introduced a bug I fixed last night.

Reply:
Um... What are you talking about?

Reply:
Fixed now. My bad, sorry. There might be similar bug for in `g4/adc.h` btw.